Program.cs

```diff
@@ -49,11 +49,21 @@ static void Main(string[] args)
 
             try
             {
-                foreach (var arg in args)
+                for (int i = 0; i < args.Length; i++)
                 {
-                    parameter = arg.Substring(0, arg.IndexOf('='));
-                    parameterValue = arg.Substring(arg.LastIndexOf('=') + 1);
```

Comment:
Since you already know that it's an out of range exception 

                    if(i + 1  < args.Length) {
                        parameter = args[i];
                        parameterValue = args[i + 1];
                        i++;
                    } else {
	                throw new IndexOutOfRangeException(ie.Message);
	            }